src/main.rs

```diff
@@ -0,0 +1,206 @@
+extern crate clap;
+extern crate unicode_normalization;
+
+use clap::{App, Arg};
+use patricia_tree::PatriciaSet;
+use std::cmp::Reverse;
+use std::collections::hash_map::DefaultHasher;
+use std::collections::{HashMap, HashSet};
+use std::fs::{metadata, File};
+use std::hash::{Hash, Hasher};
+use std::io::Read;
+use std::str;
+use unicode_normalization::UnicodeNormalization;
+
+const SPACES_FACTOR: usize = 6;
+
+fn get_raw_dict(name: String, resource_dir: &str) -> Vec<u8> {
```

Comment:
```suggestion
fn get_raw_dict(name: &str, resource_dir: &str) -> Vec<u8> {
```
If you pass `name` as a String the function consumes it (unless the type implements the `Copy` trait but I don't think `String` does). Here you probably just want a ref.

Reply:
Note that it's not a performance suggestion: passing `String` would likely move the variable because the value is consumed.